### ISSUE TITLE
Spaces in JVM arguments creates a non-working launcher

### ISSUE THREAD

**Issue by trejkaz** [beryx/badass-jlink-plugin#66]
I had a command-line argument for specifying the application's name, and the name had a space in it. When the batch file was generated, it didn't quote the parameter, resulting in the second word in the name becoming the main classname it was trying to run.

The code responsible is found here:

https://github.com/beryx/badass-jlink-plugin/blob/master/src/main/groovy/org/beryx/jlink/util/LaunchScriptGenerator.groovy#L70

**Comment by trejkaz** [comment]
Confirming that I get the same result with jpackage.

**Comment by trejkaz** [comment]
I have been bashing at this one for a few hours with not much luck, but I did figure out where the value is getting into. This is `build\jpackage\App Name\app\App Name.cfg`:

```
[Application]
app.name=Hakanai Agent
app.version=1.0
app.runtime=$APPDIR\runtime
app.identifier=org.trypticon.vrcagent
app.classpath=
app.mainmodule=org.trypticon.vrcagent/org.trypticon.vrcagent.Main

[JavaOptions]
-Dbranding.name=Hakanai
Agent
-Dbranding.version=0.0.0.1
--module-path
mods

[ArgOptions]
```

I thought maybe the quoting in `JPackageTaskImpl` was causing it, but I seem to get the same result with or without actually.

**Comment by trejkaz** [comment]
If I run the command myself, this does produce a working launcher:

```
"C:\Program Files\Java\jdk-13\bin\jpackage.exe" create-app-image --output "C:\Users\Trejkaz\Documents\IdeaProjects\Hakanai Agent Prototype\build\jpackage" --name "Hakanai Agent" --module-path "C:\Users\Trejkaz\Documents\IdeaProjects\Hakanai Agent Prototype\build\jlinkbase\jlinkjars" --module org.trypticon.vrcagent/org.trypticon.vrcagent.Main --runtime-image "C:\Users\Trejkaz\Documents\IdeaProjects\Hakanai Agent Prototype\build\image" --java-options "\"-Dbranding.name=Hakanai Agent\"" --java-options "\"-Dbranding.version=0.0.0.1\"" --win-console
```

The code _seems_ to be adding these quotes to the command-line, so I don't get why it doesn't work with the current code, nor how to fix it.

**Comment by siordache** [comment]
Fixed in release 2.12.0.
Currently, spaces in JVM arguments are processed incorrectly by jpackage. The plugin implements [a workaround](https://mail.openjdk.java.net/pipermail/core-libs-dev/2019-July/061288.html) for this issue.

**Comment by trejkaz** [comment]
Verified, with apologies for the delay. A lot of other things are going on here lately. :(
